runtime/src/bank.rs

```diff
@@ -3651,6 +3653,39 @@ impl Bank {
         }
     }
 
+    fn simulation_override_accounts(&self) -> AccountOverrides {
+        let mut account_overrides = AccountOverrides::new();
+        self.add_account_override(
+            &mut account_overrides,
+            &sysvar::slot_history::id(),
+            |account| {
+                let mut slot_history = account
+                    .as_ref()
+                    .map(|account| from_account::<SlotHistory, _>(account).unwrap())
+                    .unwrap_or_default();
+                slot_history.add(self.slot().saturating_sub(1));
+                create_account(
+                    &slot_history,
+                    self.inherit_specially_retained_account_fields(account),
+                )
+            },
+        );
+        account_overrides
+    }
+
+    fn add_account_override<F>(
```

Comment:
Probably better to inline this for now since it's only used in one spot. But at the very least, this should be renamed because at a glance, `bank.add_account_override(..)` doesn't do what I would expect.

Reply:
yeah, this function is a bit confusing.
I also think could be better to inline this code.